src/EncompassRest/Loans/LoanFieldsCustomizations.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using EncompassRest.Utilities;
+using Newtonsoft.Json.Serialization;
+
+namespace EncompassRest.Loans
+{
+    public sealed partial class LoanFields
+    {
+        private readonly Loan _loan;
+
+        public object this[string fieldId]
+        {
+            get
+            {
+                Preconditions.NotNullOrEmpty(fieldId, nameof(fieldId));
+
+                fieldId = fieldId.ToUpper();
+                if (fieldId.StartsWith("CX.", StringComparison.Ordinal))
+                {
+                    var customField = _loan.CustomFields.FirstOrDefault(f => string.Equals(fieldId, f.FieldName, StringComparison.OrdinalIgnoreCase));
+                    if (customField != null)
+                    {
+                        if (customField.DateValue.HasValue)
+                        {
+                            return customField.DateValue;
+                        }
+                        if (customField.NumericValue.HasValue)
+                        {
+                            return customField.NumericValue;
+                        }
+                        return customField.StringValue;
+                    }
+                    return null;
+                }
+                else
+                {
+                    var finalProperty = GetFinalProperty(fieldId, out var target);
+                    var result = finalProperty.ValueProvider.GetValue(target);
+                    var propertyTypeInfo = finalProperty.PropertyType.GetTypeInfo();
+                    if (propertyTypeInfo.IsGenericType && !propertyTypeInfo.IsGenericTypeDefinition)
+                    {
+                        var genericTypeDefinition = propertyTypeInfo.GetGenericTypeDefinition();
+                        if (genericTypeDefinition == TypeData.OpenStringEnumValueType || genericTypeDefinition == TypeData.OpenNaType)
+                        {
+                            result = result.ToString();
+                        }
+                    }
+                    return result;
+                }
+            }
+            set
+            {
+                Preconditions.NotNullOrEmpty(fieldId, nameof(fieldId));
+
+                fieldId = fieldId.ToUpper();
+                if (fieldId.StartsWith("CX.", StringComparison.Ordinal))
+                {
+                    var customField = _loan.CustomFields.FirstOrDefault(f => string.Equals(fieldId, f.FieldName, StringComparison.OrdinalIgnoreCase));
+                    if (customField == null)
+                    {
+                        customField = new CustomField { FieldName = fieldId };
```

Comment:
this should check if the field is available in the schema.  if not, throw argument exception.

Reply:
What if they're just updating a field on a loan that hasn't been retrieved? We would have no idea what custom fields are available.

Reply:
it would still show up in the schema call.  possibly use getSchemaField ?

Reply:
You cannot retrieve custom fields. Besides, that's an asynchronous call.

Reply:
add boolean option for createIfNotExist?  there has to be some "validation" that what you are writing is valid.